src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectSystemCache.cs

```diff
@@ -22,7 +22,7 @@ internal class ProjectSystemCache : IProjectSystemCache, IDisposable
         // 0 - Cache is clean
         // 1 - Cache is dirty
         private int _isCacheDirty = 0;
-        private readonly Dictionary<string, CacheEntry> _primaryCache = new Dictionary<string, CacheEntry>();
+        private readonly Dictionary<string, CacheEntry> _primaryCache = new Dictionary<string, CacheEntry>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Discussed offline. 

This runs on windows only. Windows is case insensitive  ve, and the other caches are case insensitive already, so it's likely not a problem. 

The concern would be, is this hiding some inconsistency that could bite us in other ways. Is it our APIs? project-system etc?

This might be easiest time to figure that out. 
Given that ProjectSystemCache is a MEF type, maybe we try importing https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio.Common/OutputConsoleLogger.cs and logging when we see the inconsistency. 

 In isolation, this change is likely good, but maybe there's an easy way to confirm 100%.